### ISSUE TITLE
Can request the Triton Inference Server by using the 'request package' in python rather than 'client library'?

### ISSUE THREAD

**Issue by DataXujing** [triton-inference-server/server#2118]
Can request the Triton Inference Server by using the 'request package' in python rather than 'client library'?

like:
```
import request
res = requests.post(url, params=params）
print(res.text)
```

**Comment by deadeyegoodwin** [comment]
Yes, you can format and send your own HTTP requests, you just need to follow the HTTP/REST protocol https://github.com/triton-inference-server/server/blob/master/docs/inference_protocols.md.

Note that you will likely have lower performance than if you use the client library unless you take advantage of the binary data extension.

**Comment by thinkinds** [comment]
But when I use request.post to call the Triton Inference Server,  I get error messeages like: Infer failed: inference request batch-size must be <= 1 for 'my_model_name', but actually I send only 1 request, in which  a  frame of video data included.
The same method I used with Tf-serving and worked. Why it failed with  the Triton Inference Server?

**Comment by tanmayv25** [comment]
The first dimension of shape of provided tensor is treated at the batch dimension even if you send a single request. Try this [1, shape].

**Comment by tanmayv25** [comment]
This is already documented in the protocol.

> The $request_input JSON describes an input to the model. If the input is batched, the shape and data must represent the full shape and contents of the entire batch.


Source: https://github.com/kubeflow/kfserving/blob/master/docs/predict-api/v2/required_api.md#request-input

**Comment by thinkinds** [comment]
> This is already documented in the protocol.
> 
> > The $request_input JSON describes an input to the model. If the input is batched, the shape and data must represent the full shape and contents of the entire batch.
> 
> Source: https://github.com/kubeflow/kfserving/blob/master/docs/predict-api/v2/required_api.md#request-input

I must admit I've read this, but didn't understand it. Maybe an example would help a lot.

**Comment by david-waterworth** [comment]
Yes that sentence is incomplete in my opinion - it says "if the input is batched" and makes not mention of what should happen when the input is not batched, nor that in fact the input essentially always has to be batched with batch_size>=1.
